packages/apollo-codegen-swift/src/language.ts

```diff
@@ -55,38 +43,263 @@ const reservedKeywords = new Set(['associatedtype', 'class', 'deinit', 'enum', '
   'indirect', 'lazy', 'left', 'mutating', 'none', 'nonmutating', 'optional',
   'override', 'postfix', 'precedence', 'prefix', 'Protocol', 'required', 'right',
   'set', 'Type', 'unowned', 'weak', 'willSet']);
+const reservedMemberKeywords = new Set(["self", "Type", "Protocol"]);
+
+/**
+ * A class that represents Swift source.
+ *
+ * Instances of this type will not undergo escaping when used with the `swift` template tag.
+ */
+export class SwiftSource {
+  source: string;
+  constructor(source: string) {
+    this.source = source;
+  }
+
+  /**
+   * Returns the input wrapped in quotes and escaped appropriately.
+   * @param string The input string, to be represented as a Swift string.
+   * @param trim If true, trim the string of whitespace and join into a single line.
+   * @returns A `SwiftSource` containing the Swift string literal.
+   */
+  static string(string: string, trim: boolean = false): SwiftSource {
+    if (trim) {
+      string = string
+        .split(/\n/g)
+        .map(line => line.trim())
+        .join(" ");
+    }
+    return new SwiftSource(
+      `"${string.replace(/[\0\\\t\n\r"]/g, c => {
```

Comment:
Would you mind throwing in a comment for the regex-impared explaining what's being replaced here?

Reply:
To be honest, this is a pretty simple regex. It's just matching any of 6 characters. I suppose I could say something like `// Match any character that needs escaping in Swift strings` but I thought that was inferable from context.

Reply:
It is, but I think maybe what threw me is I'm not really sure what's being escaped with `\0`, or why a single quote isn't being escaped. The other bits are pretty straightforward.

Reply:
Technically all I really need to escape is `\n`, `\r`, `"`, and `\\`, but I included `\0` (the NUL character) and `\t` because Swift has defined escapes for those and that makes the source easier to read. Single quote isn't escaped because it's not a special character in a Swift string literal.

Reply:
OK - maybe if there's someplace where those are defined it might be worth throwing in a comment link to it. Otherwise, this is fine.

Reply:
Good idea, I can put in a link to the Swift language reference.